src/main/java/duke/command/DeleteCommand.java

```diff
@@ -0,0 +1,47 @@
+package duke.command;
+
+import duke.Duke;
+import duke.storage.Storage;
+import duke.task.Task;
+import duke.tasklist.TaskList;
+import duke.ui.Ui;
+
+/**
+ * Represents a user command that deletes a task from the task list.
+ */
+public class DeleteCommand extends Command {
+    private String userInput;
+
+    public DeleteCommand(String userInput) {
+        this.userInput = userInput;
+    }
+
+    public boolean isExit() {
+        return false;
+    }
+
+    /**
+     * Checks for the number specified by the user, and deletes the task at that
+     * number accordingly.
+     *
+     * @param tasks List of <code>Task</code> objects.
+     * @param ui Ui object created by Duke.
+     * @param storage Storage object created by Duke.
+     */
+    public void execute(TaskList tasks, Ui ui, Storage storage) {
+        String fileString = tasks.listToString();
+        int taskNumber = Integer.parseInt(userInput.substring(7));
```

Comment:
Maybe extract the magic number 7 out into a constant variable? Naming it would make what the 7 stands for clearer.